pkg/ctl/enable/profile.go

```diff
@@ -4,199 +4,145 @@ import (
+	})
 
-	// Create the profile generator. It will output the processed templates into a new "/base" directory into the user's repo
-	usersRepoName, err := git.RepoName(opts.gitOptions.URL)
+	err = gitClient.CloneRepoInPath(
+		usersRepoDir,
+		git.CloneOptions{
+			URL:       opts.gitOptions.URL,
+			Branch:    opts.gitOptions.Branch,
+			Bootstrap: true,
+		},
+	)
 	if err != nil {
 		return err
 	}
-	dir, err := ioutil.TempDir("", usersRepoName)
-	logger.Debug("Directory %s will be used to clone the configuration repository and install the profile", dir)
-	usersRepoDir := filepath.Join(dir, usersRepoName)
-	profileOutputPath := filepath.Join(usersRepoDir, "base")
 
 	profile := &gitops.Profile{
-		Processor: processor,
-		Path:      profileOutputPath,
+		Processor: &fileprocessor.GoTemplateProcessor{
+			Params: fileprocessor.NewTemplateParameters(cmd.ClusterConfig),
+		},
+		Path: profileOutputPath,
 		GitOpts: git.Options{
 			URL:    profileRepoURL,
-			Branch: opts.profileRevision,
+			Branch: opts.profileOptions.Revision,
 		},
-		GitCloner: git.NewGitClient(git.ClientParams{}),
-		FS:        afero.NewOsFs(),
-		IO:        afero.Afero{Fs: afero.NewOsFs()},
+		GitCloner: git.NewGitClient(git.ClientParams{
+			PrivateSSHKeyPath: opts.gitOptions.PrivateSSHKeyPath,
+		}),
+		FS: afero.NewOsFs(),
+		IO: afero.Afero{Fs: afero.NewOsFs()},
 	}
 
-	// A git client that operates in the user's repo
-	gitClient := git.NewGitClient(git.ClientParams{
-		PrivateSSHKeyPath: opts.gitPrivateSSHKeyPath,
-	})
-
-	gitOps := gitops.Applier{
-		UserRepoPath:     usersRepoDir,
-		UsersRepoOpts:    opts.gitOptions,
-		GitClient:        gitClient,
-		ProfileGenerator: profile,
-		FluxInstaller:    fluxInstaller,
-		ClusterConfig:    cmd.ClusterConfig,
-		QuickstartName:   opts.profileNameArg,
+	err = profile.Generate(context.Background())
+	if err != nil {
+		return errors.Wrap(err, "error generating profile")
 	}
 
-	if err = gitOps.Run(context.Background()); err != nil {
+	// Git add, commit and push component files in the user's repo
+	if err = gitClient.Add("."); err != nil {
 		return err
 	}
-	os.RemoveAll(dir) // Only clean up if the command completely successfully, for more convenient debugging.
-	return nil
-}
-
-func repoURLForQuickstart(quickstartArgument string) (string, error) {
-	if git.IsGitURL(quickstartArgument) {
-		return quickstartArgument, nil
-	}
-	if quickstartArgument == "app-dev" {
-		return "https://github.com/weaveworks/eks-quickstart-app-dev", nil
+	commitMsg := fmt.Sprintf("Add %s quickstart components", opts.profileOptions.Name)
+	if err = gitClient.Commit(commitMsg, opts.gitOptions.User, opts.gitOptions.Email); err != nil {
+		return err
 	}
-	if quickstartArgument == "appmesh" {
-		return "https://github.com/weaveworks/eks-appmesh-profile", nil
+	if err = gitClient.Push(); err != nil {
+		return err
 	}
-	return "", fmt.Errorf("invalid URL or unknown Quick Start profile %s ", quickstartArgument)
+
+	profile.DeleteClonedDirectory()
```

Comment:
Now that the clone is done outside of the `profile` it may make more sense to delete the directory from here instead. What do you think?

Reply:
Given:
```
Cloning into '/var/folders/24/d3mml6bn20nftpt91cfldq1h0000gn/T/my-gitops-repo-053697054'...
[...]
Cloning into '/var/folders/24/d3mml6bn20nftpt91cfldq1h0000gn/T/quickstart-041818341'...
```
we should probably have this instead:
```suggestion
	profile.DeleteClonedDirectory()
	os.RemoveAll(usersRepoDir)
```
Or did you have something else in mind?

Reply:
FWIW, the above suggestion is what I just added in `enable profile: delete temp. clone of user's config repo`.

Reply:
I would remove the `profile.DeleteClonedDirectory()` and have:
```
os.RemoveAll(usersRepoDir)
os.RemoveAll(quickstartRepoDir....)
```
That method "DeleteclonedDirectory" was there before because the `profile` created it, but now it's created outside right? So that's why I think the delete logic should move out too.

Reply:
Oh, I was confused. `profile.DeleteClonedDirectory()` actually deletes the quickstart profile repo right? Which is cloned by `profile` earlier so that suggestion you proposed is correct!

Reply:
Yes, it does delete it. It _is_ confusing: it is really not explicit that it not only takes the directory of the users' repository in order to copy generated files there, but also clones the profile in another temporary directory. Indeed, `profile.Generate`calls
https://github.com/weaveworks/eksctl/blob/38daf7dec34127ef63c0da6bc2144fb091f5b20c/pkg/gitops/profile.go#L38-L46 which calls
https://github.com/weaveworks/eksctl/blob/38daf7dec34127ef63c0da6bc2144fb091f5b20c/pkg/git/git.go#L91-L98

Reply:
Also, I undid the above suggestion as it led to duplicate code as I already did the equivalent in https://github.com/weaveworks/eksctl/pull/1509#discussion_r342503348
```diff
$ git diff HEAD~1..HEAD
diff --git a/pkg/ctl/enable/profile.go b/pkg/ctl/enable/profile.go
index 1916ff89..5ba345ac 100644
--- a/pkg/ctl/enable/profile.go
+++ b/pkg/ctl/enable/profile.go
@@ -146,5 +146,6 @@ func doEnableProfile(cmd *cmdutils.Cmd, opts *ProfileOptions) error {
 
        profile.DeleteClonedDirectory()
        os.RemoveAll(usersRepoDir)
+       os.RemoveAll(usersRepoDir)
        return nil
 }
```